drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -640,6 +640,7 @@ int usb_dc_ep_enable(const uint8_t ep)
 	/* clear buffer state (EP0 starts with PID=1 for setup phase) */
 
 	*ep_state->buf_ctl = (USB_EP_GET_IDX(ep) == 0 ? USB_BUF_CTRL_DATA1_PID : 0);
+	ep_state->next_pid = (USB_EP_GET_IDX(ep) == 0 ? 1 : 0);
```

Comment:
This is a bit confusing (also the comment above, DATA0 PID follows SETUP PID), maybe it works because setup packet can always be received and driver below does not call udc_rpi_start_xfer() for control out endpoint and next PID is explicitly updated in udc_rpi_handle_setup(). Can you change these two lines, 642 and 643, to
```c

	*ep_state->buf_ctl = USB_BUF_CTRL_DATA0_PID;
	ep_state->next_pid = 0;
```
and test it?

Reply:
Okay, I changed it according to your suggestion and it works on my side.

Reply:
But do **you** think it is the right way and makes sense? Not that you changed it just because I suggested it.

Reply:
It makes sense to me. As you said, next_pid for control is forced in udc_rpi_handle_setup. I had another close look at the code and I'm confident that this is the right way.